packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -365,17 +365,38 @@ export class BackendInitializer {
             }
             await tree.parallelTopologicalTraversal(
               async ({ moduleId, moduleInit }) => {
-                const moduleDeps = await this.#getInitDeps(
-                  moduleInit.init.deps,
-                  pluginId,
-                  moduleId,
-                );
-                await moduleInit.init.func(moduleDeps).catch(error => {
-                  throw new ForwardedError(
-                    `Module '${moduleId}' for plugin '${pluginId}' startup failed`,
-                    error,
+                const isModuleBootFailurePermitted =
+                  this.#getPluginModuleBootFailurePredicate(
+                    pluginId,
+                    moduleId,
+                    rootConfig,
                   );
-                });
+
+                try {
+                  const moduleDeps = await this.#getInitDeps(
+                    moduleInit.init.deps,
+                    pluginId,
+                    moduleId,
+                  );
+                  await moduleInit.init.func(moduleDeps).catch(error => {
+                    throw new ForwardedError(
+                      `Module '${moduleId}' for plugin '${pluginId}' startup failed`,
+                      error,
+                    );
+                  });
+                } catch (error: unknown) {
+                  assertError(error);
+                  if (isModuleBootFailurePermitted) {
+                    initLogger.onPermittedPluginModuleFailure(
+                      pluginId,
+                      moduleId,
+                      error,
+                    );
+                  } else {
+                    initLogger.onPluginModuleFailed(pluginId, moduleId, error);
+                    throw error;
+                  }
+                }
```

Comment:
this could be brought into it's own method that could be shared between plugins, and modules. Not sure if that's the desired behavior. Happy to take it that direction if that is what is deemed best.